pkg/daemon/manager/manager.go

```diff
@@ -75,6 +78,72 @@ func (m *manager) DeleteService(name string) error {
 	return winSvc.Delete()
 }
 
+func (m *manager) DeleteServiceAndDependents(name string) error {
+	existingSvcs, err := m.GetServices()
+	if err != nil {
+		return errors.Wrap(err, "could not determine existing Windows services")
+	}
+	// Nothing to do if it already does not exist
+	if _, present := existingSvcs[name]; !present {
+		return nil
+	}
+
+	winSvcObj, err := m.OpenService(name)
+	if err != nil {
+		return errors.Wrapf(err, "unable to open %s service handler", name)
+	}
+	defer winSvcObj.Close()
+
+	status, err := winSvcObj.Query()
+	if err != nil {
+		return errors.Wrapf(err, "unable to query %s service status", name)
+	}
+	if status.State != svc.Stopped {
+		// Ensure no dependent services are running before we can safely stop this one
+		if err := m.removeDependentServices(name, existingSvcs); err != nil {
+			return err
+		}
+	}
+	klog.Infof("removing service %s", name)
```

Comment:
Instead of logging here, I recommend returning a []string of services that have been removed. This will solve the issue presented [here](https://github.com/openshift/windows-machine-config-operator/pull/1154/commits/a5052500d56f0cd3ae07a9295bb77747291926a2#r947221980)

Reply:
This slice can be appended to as you recurse

Reply:
Logging a concise list of the removed services, instead of having individual log lines for each service is going to help the WMCO error logs to not be a mess.